### ISSUE TITLE
Choose and define unambiguous terms for different raster resampling and rescaling operations

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-api-poc#43]
Taken over from the glossary repository:

@mkadunc commented on 8 Dec 2017:
> There's some confusion in the image processing and raster data world about terms related to resampling or rescaling of images.
> 
> I support the current description in the glossary where aggregation is separate from general resampling. Constraints regarding uniqueness of elements makes it possible to view the `aggregation`
 operation as a mapping of an aggregate operator (or a reducer) over the blocks that make up some partition of the collection ([in mathematical sense](https://en.wikipedia.org/wiki/Partition_of_a_set), analogous to `SELECT ... GROUP BY ...` in SQL).
> 
> Resampling-related operations on raster data are called many names e.g. resampling, upsampling, downsampling, interpolation, decimation, upscaling, downscaling ([from wikipedia](https://en.wikipedia.org/wiki/Sample-rate_conversion)), warping etc.
> 
> If we define a raster as a regular discrete sampling of some (unknown) continuous variable, all such operations can be modeled as a combination of two operations:
> * (optional) transform of the continuous variable, usually by convolving with a low-pass-like window function
> * sampling of the (interpolated and/or transformed) continuous variable at the target grid locations
> In this framework, it is quite clear that upscaling an image by a factor of 2 (usually done with bicubic interpolation) is a completely different operation than downscaling an image by a factor of 2 (usually done with a box or gaussian window) - the former preserves the continuous variable and its meaning (the raster can still be understood as the sampling of the same variable as before), while the latter reduces the scale of the variable (in the sense of [Scale Space](https://en.wikipedia.org/wiki/Scale_space)), causing the raster to represent a different variable.
> 
> The following two attributes of a raster are enough to clearly categorize most of the frequently used resampling operations:
> * spatial resolution: effective maximum resolution of the continuous variable sampled by the raster (defined by: sensor spatial resolution; [MTF](https://www.edmundoptics.com/resources/application-notes/optics/introduction-to-modulation-transfer-function/) of the process that produced source data; combined width of low-pass filtering suring processing etc.) ~ analogous to https://en.wikipedia.org/wiki/Image_resolution , https://en.wikipedia.org/wiki/Optical_resolution) and related to [image scale](https://en.wikipedia.org/wiki/Scale_space)
> * GSD (ground sampling distance) / pixel size / grid size : distance between grid elements in a (spatial) dimension
> Operations, often used to process raster imagery, might then be described as:
> * downscaling: increases the GSD and proportionately lowers spatial resolution (as if applying a low-pass window)
> * upsampling: decreases GSD and maintains the same spatial resolution (uses interpolation)
> * warping: changing the raster grid, usually with a non-affine transform, while approximately preserving both the spatial resolution and GSD of the data (uses interpolation)
> * blurring: lowering the spatial resolution while preserving the raster grid (and GSD) (uses low-pass convolution)
> * super-resolving / PAN-sharpening : decreases GSD and increases spatial resolution (inference of missing high-frequency information from auxiliary datasets or statistics)
> 
> **TL; DR:** In OpenEO, we will encounter the need to use resampling-related terms mostly when defining warping, downscaling and upsampling processes (in case the standard decides to deal with this). To avoid ambiguity, I suggest to treat these three types of processes separately, and provide separate code-lists for the methods to be used in them (interpolation methods for upsampling and scale-preserving warp; window-averaging methods for downscaling).
> 
> P.S.: If there's a consistent definition of these concepts and such operations in any of the existing standards that we're looking at (WCPS / array SQL...), we might be able to adopt their terminology and best practices.

@edzer commented on 8 Dec 2017:
> Thanks! Relevant OGC WCS standards seem those addressing
> * [scaling](https://portal.opengeospatial.org/files/12-039), and
> * [interpolation](https://portal.opengeospatial.org/files/12-049)
